cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceService.java

```diff
@@ -171,6 +188,15 @@ public void create(Id.Namespace namespace, String name, DatasetInstanceConfigura
     enableExplore(newInstance, props);
   }
 
+  private void createIfNotExists(Id.Namespace namespace, String name,
+                                 DatasetInstanceConfiguration props) throws Exception {
+    try {
+      create(namespace, name, props);
```

Comment:
It might be a good idea to do a check for existence before trying to create like you do in `RemoteDatasetFramework`.